### PR TITLE
s/EntityManager/EntityManagerInterface in type-hints

### PR DESCRIPTION
**Pull request by Ocramius**

### CODE REVIEW

File: lib/Doctrine/ORM/Id/AbstractIdGenerator.php [hunk 1]
```diff
@@ -19,19 +19,19 @@
...
      * Generates an identifier for an entity.
      *
-     * @param \Doctrine\ORM\EntityManager  $em
+     * @param EntityManagerInterface       $em
      * @param \Doctrine\ORM\Mapping\Entity $entity
      *
      * @return mixed
      */
-    abstract public function generate(EntityManager $em, $entity);
+    abstract public function generate(EntityManagerInterface $em, $entity);
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Id/AbstractIdGenerator.php]: Not actually a BC break, as it is in ORM internals, but I will have to document the change

**Comment by fprochazka** [comment]
Just a quick note: the typehint is also changed in `ClassMetadataFactory::setEntityManager` - so that should also be documented.

File: lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php [hunk 2]
```diff
@@ -70,9 +70,9 @@ class ClassMetadataFactory extends AbstractClassMetadataFactory
     private $embeddablesActiveNesting = array();
 
     /**
-     * @param EntityManager $em
+     * @param EntityManagerInterface $em
      */
-    public function setEntityManager(EntityManager $em)
+    public function setEntityManager(EntityManagerInterface $em)
```
Comment (fprochazka) [inline lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php]: This was a BC for me. I was overriding this method to get to the EM in metadata factory.

Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php]: Will document it

**Pull request merged**